pkgs/development/coq-modules/bignums/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, coq }:
+
+let rev_and_sha = {
+  "8.6" = {
+    rev = "v8.6.0";
+    sha256 = "0553pcsy21cyhmns6k9qggzb67az8kl31d0lwlnz08bsqswigzrj";
+  };
+  "8.7" = {
+    rev = "V8.7.0";
+    sha256 = "11c4sdmpd3l6jjl4v6k213z9fhrmmm1xnly3zmzam1wrrdif4ghl";
+  };
+};
+in
+
+if ! (rev_and_sha ? "${coq.coq-version}") then
+  throw "bignums is not available for Coq ${coq.coq-version}"
+else with rev_and_sha."${coq.coq-version}";
+
+stdenv.mkDerivation rec {
+
+  name = "coq${coq.coq-version}-bignums";
+
+  src = fetchFromGitHub {
+    owner = "coq";
+    repo = "bignums";
+    inherit rev sha256;
```

Comment:
Does this build? I don't see `rev` and `sha256` in the parent dictionary.

Reply:
Yes it builds, the rev and sha256 are defined in an attribute set above that is imported using `with`.

Reply:
Ah, I see. The indentation is a little strange, it doesn't make it clear that the `mkDerivation` is a sub-expression of the if-else clause. Not a huge deal, though.